### PR TITLE
[WIP][Workflow] Changed initial_places to initial_marking, added property instead of type

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #30656
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/11211

ping @lyrixx

### CODE REVIEW

**Comment by lyrixx** [comment]
@HeahDude Hi. Thanks for the PR. What is missing? Do you think you will be able to finish it before the feature freeze?

**Comment by HeahDude** [comment]
Hey!

>What is missing? 

I need to fix the CI and get your approval :). Is the naming ok? I'm not sure about the deprecation messages, I surely need to update the changelog and upgrade files about them and the `initial_marking`.

>Do you think you will be able to finish it before the feature freeze?

Sure, if you ask me, I would love to have it merged for tomorrow :).

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 1]
```diff
@@ -295,12 +296,14 @@
...
         <xsd:attribute name="type" type="marking_store_type" />
         <xsd:attribute name="service" type="xsd:string" />
+        <xsd:attribute name="property" type="xsd:string" />
     </xsd:complexType>
 
     <xsd:simpleType name="marking_store_type">
         <xsd:restriction base="xsd:string">
             <xsd:enumeration value="multiple_state" />
             <xsd:enumeration value="single_state" />
+            <xsd:enumeration value="method" />
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: This is a kind of bug fix

**Comment by vinise** [comment]
`->enumNode('type')
 ->values(['workflow', 'state_machine'])
 ->defaultValue('state_machine')`

Missing "method" value ;)

File: UPGRADE-4.3.md [hunk 2]
```diff
@@ -183,19 +178,22 @@ Workflow
    ```yaml
    framework:
        workflows:
+           type: workflow
            article:
                marking_store:
                    type: multiple
+                   arguments: states
```
Comment (lyrixx) [inline UPGRADE-4.3.md]: I would stick to the default value instead

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 3]
```diff
@@ -637,6 +637,12 @@ private function registerWorkflowConfiguration(array $config, ContainerBuilder $
                 foreach ($workflow['marking_store']['arguments'] as $argument) {
                     $markingStoreDefinition->addArgument($argument);
                 }
+                if ('method' === $workflow['marking_store']['type']) {
+                    $markingStoreDefinition->setArguments([
+                        'state_machine' === $type, //single state
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: This assumption is wrong. You can pre-configure with this code, but The value should be overrideable by the enduser.
It **is** possible to have a workflow with a "single state" marking store

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 4]
```diff
@@ -637,6 +637,12 @@ private function registerWorkflowConfiguration(array $config, ContainerBuilder $
...
+                        $workflow['marking_store']['property'] ?? 'marking',
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: This should be the default value in the Configuration (not the extension)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 5]
```diff
@@ -270,7 +270,7 @@
 
     <xsd:complexType name="workflow">
         <xsd:sequence>
-            <xsd:element name="initial-place" type="xsd:string" minOccurs="0" maxOccurs="unbounded" />
+            <xsd:element name="initial-marking" type="xsd:string" minOccurs="0" maxOccurs="unbounded" />
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: both element should be here to support the BC, no ?

**Review by lyrixx — COMMENTED** [review]
Thanks, I left few comments

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 3 above)
Comment (lyrixx) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Actually, this is useless. :+1: with your patch

**Comment by lyrixx** [comment]
@HeahDude I could finish this PR. Are you OK with that?

File: UPGRADE-4.3.md (hunk 2 above)
Comment (lyrixx) [reply UPGRADE-4.3.md]: Actually this is nice. It's better to fully understand if the property will contain a single place, or many places :+1:

**Comment by HeahDude** [comment]
Yes please do!

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd (hunk 5 above)
Comment (HeahDude) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Yes

**Comment by lyrixx** [comment]
Thank you very much for you initial work on this PR.
I finished it in #30890

**Pull request closed without merging**